lib/Conversion/TorchToTcp/Misc.cpp

```diff
@@ -84,7 +86,8 @@ class ConvertAtenBroadcastToOp : public OpConversionPattern<AtenBroadcastToOp> {
       int64_t staticDimSize;
       if (i < newLeadingDims ||
           !matchPattern(newDimSize, m_TorchConstantInt(&staticDimSize)) ||
-          staticDimSize != -1) {
+          (staticDimSize != -1 &&
+           staticDimSize != inputShape[i - newLeadingDims])) {
```

Comment:
perhaps add a short comment above explaining the rationale

Reply:
+1 to this.

How about adding some lambda function that describes what it is actually checking, say something like `isDimBroadcasted` for the new condition you added?